### DISCUSSION TITLE
An existing connection was forcibly closed by the remote host

### DISCUSSION THREAD

**Discussion by karthickprakash** [CoreWCF/CoreWCF#1485] [Q&A] [answered]
Hi @mconnew 

I am encountering an issue while transferring a 200 MB file from a client to a CoreWCF service using BasicHttpBinding. I'm receiving the following error message:

```c#
An error occurred while receiving the HTTP response to
http://localhost:5000/FileManagementService.svc.
This could be due to the service endpoint binding not using the HTTP protocol. This could also be due to an HTTP request context being aborted by the server (possibly due to the service shutting down). See server logs for more details
``` 

To address this, I've attempted to increase the **MaxReceivedMessageSize** property of the **BasicHttpBinding**, but it doesn't seem to be resolving the issue. 

![image](https://github.com/user-attachments/assets/3c7d9b7c-3c15-44af-971f-d3ec7ab17df6)

While I can successfully upload 10 MB, 20 MB files, the problem arises with 30 MB files and above.

Could you please suggest what needs to be done to get this fixed?

**Comment by mconnew** [comment, marked as answer]
You are probably hitting a Kestrel request size limitation.
```c#
public void ConfigureServices(IServiceCollection services)
{
    services.Configure<KestrelServerOptions>(options =>
    {
        options.Limits.MaxRequestBodySize = 250 * 1024 * 1024; // 250 MB limit
    });
}
```
You could also insert this code before `app.UseServiceModel` to change it at the point of receiving a request. This is generally useful if you are hosting other things and only want to increase the limit for your CoreWCF service:
```c#
app.Map("/FileManagementService.svc", serviceApp => 
{
    serviceApp.Use(async (context, next) =>
    {
        var httpMaxRequestBodySizeFeature = context.Features.Get<IHttpMaxRequestBodySizeFeature>();
        if (httpMaxRequestBodySizeFeature is not null)
        {
            httpMaxRequestBodySizeFeature.MaxRequestBodySize = 250 * 1024 * 1024; // 250 MB limit
        }
        await next(context);
    });
    serviceApp.UseServiceModel(builder =>
    {
        /* Configure your CoreWCF service */
    });
});
```

**Comment by karthickprakash** [comment]
Thank you @mconnew for the answer. I am able to send the files now.
